pyuvdata/uvdata/mir_parser.py

```diff
@@ -10,1082 +10,6383 @@
+        # we will fix this, but for now, we triage the autos here. Note that if we
+        # already have the auto_data loaded, we can bypass this step.
+        if load_auto:
+            self.auto_data = self.read_data(
+                "auto", return_vis=False, use_mmap=use_mmap, read_only=read_only
+            )
+
+    def unload_data(self, unload_vis=True, unload_raw=True, unload_auto=True):
+        """
+        Unload data from the MirParser object.
+
+        Unloads the data-related attributes from memory, if they are loaded. Because
+        these attributes can be formidible in size, this operation will substantially
+        reduce the memory footprint of the MirParser object.
+
+        Parameters
+        ----------
+        unload_vis : bool
+            Unload the visibilities stored in the `vis_data` attribute, if loaded.
+            Default is True.
+        unload_raw : bool
+            Unload the raw visibilities stored in the `raw_data` attribute, if loaded.
+            Default is True.
+        unload_auto : bool
+            Unload the auto-correlations stored in the `auto_data` attribute, if loaded.
+            Default is True.
+        """
+        if unload_vis:
+            self.vis_data = None
+            self._tsys_applied = False
+        if unload_raw:
+            self.raw_data = None
+        if unload_auto:
+            self.auto_data = None
+
+    def _update_filter(self, update_data=None):
+        """
+        Update MirClass internal filters for the data.
+
+        Note that this is an internal helper function which is not for general user use,
+        but instead is part of the low-level API for the MirParser object. Updates
+        the masks of the various MirMetaData objects so that only records with entries
+        across _all_ the metadata objects are included. Typically used after issuing
+        a `select` command to propagate masks to objects that did not immediately have
+        matching selection criteria based on what the user provided.
+
+        update_data : bool
+            If set to True, will read in data from disk after selecting records. If
+            set to False, data attributes (e.g., `vis_data`, `raw_data`, `auto_data`)
+            will be unloaded. If set to True, data attributes will be reloaded, based
+            on what had been previously.  Default is to downselect the data from that
+            previously unloaded if possible, otherwise unload the data.
+        """
+        mask_update = False
+        # Start by cascading the filters up -- from largest metadata tables to the
+        # smallest. First up, spec win -> baseline
+        if not np.all(self.sp_data.get_mask()):
+            mask_update |= self.bl_data.set_mask(header_key=set(self.sp_data["blhid"]))
+
+        # Now do baseline -> antennas. Special handling required because of the
+        # lack of a unique index key for this table.
+        if not np.all(self.bl_data.get_mask()):
```

Comment:
On second glance, the logic here wasn't quite right, in part because I had just folded in the auto-correlation stuff. It's improved now, making a call to `MirParser.eng_data.get_mask` (which is both the intended route, and is faster on account of  fewer records to parse through). Fixed in c217b41d.